compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs

```diff
@@ -40,10 +40,7 @@ renderRst env = \case
         ] ++
         [ T.concat
             [ "   "
-            , unModulename moduleName
-            , " <"
             , T.pack (moduleNameToFileName moduleName)
-            , ">"
```

Comment:
@bame-da I don't understand the intention here, this removes the links from the `daml damlc docs`-generated TOC, which seems unrelated to the problem you're trying to solve.

Reply:
I took the freedom to undo this change

Reply:
No, please don't.

A toc entry of the format

`Custom title <doc path>`

is only needed if you want the text in the ToC to be different to the page header.

If you want the text in the ToC to just be the page header, you can just have `doc path` as the ToC entry. And that's my aim here.

Reply:
I see now, thanks for explaining!

Reply:
the part that worries me is that I'm not sure how this would interact with modules that have multiple components in their names e.g. `DA.Action`; I'll add some tests for that